test/compiler/inline.jl

```diff
@@ -1069,3 +1069,18 @@ end
 @test fully_eliminated() do
     issue41694(2)
 end
+
+let m = Module()
+    @eval m begin
+        global x::Int = 0
+        function f()
+            global x = 0
+            while x<10
+                x += 1
+            end
+            x
+        end
+    end
+    src = code_typed1(m.f)
+    @test count(x -> isa(x, Core.PiNode), src.code) == 0
+end
```

Comment:
There should not be a new module created in tests, unless essential (and other style nits)
```suggestion
global x44200::Int = 0
function f44200()
    global x = 0
    while x < 10
        x += 1
    end
    x
end
let src = code_typed1(m.f)
    @test count(x -> isa(x, Core.PiNode), src.code) == 0
end
```